cns/restserver/restserver.go

```diff
@@ -1961,3 +1969,44 @@ func (service *HTTPRestService) unpublishNetworkContainer(w http.ResponseWriter,
 	err = service.Listener.Encode(w, &response)
 	logger.Response(service.Name, response, response.Response.ReturnCode, ReturnCodeToString(response.Response.ReturnCode), err)
 }
+
+func logNCSnapshot(createNetworkContainerRequest cns.CreateNetworkContainerRequest) {
+	var aiEvent = aitelemetry.AiEvent{
+		EventName:  "CNSNCSnapshot",
+		Properties: make(map[string]string),
+		ResourceID: createNetworkContainerRequest.NetworkContainerid,
+	}
+
+	aiEvent.Properties["IPConfiguration"] = fmt.Sprintf("%+v", createNetworkContainerRequest.IPConfiguration)
+	aiEvent.Properties["LocalIPConfiguration"] = fmt.Sprintf("%+v", createNetworkContainerRequest.LocalIPConfiguration)
+	aiEvent.Properties["PrimaryInterfaceIdentifier"] = createNetworkContainerRequest.PrimaryInterfaceIdentifier
+	aiEvent.Properties["MultiTenancyInfo"] = fmt.Sprintf("%+v", createNetworkContainerRequest.MultiTenancyInfo)
+	aiEvent.Properties["CnetAddressSpace"] = fmt.Sprintf("%+v", createNetworkContainerRequest.CnetAddressSpace)
+	aiEvent.Properties["AllowNCToHostCommunication"] = fmt.Sprintf("%t", createNetworkContainerRequest.AllowNCToHostCommunication)
+	aiEvent.Properties["AllowHostToNCCommunication"] = fmt.Sprintf("%t", createNetworkContainerRequest.AllowHostToNCCommunication)
+	aiEvent.Properties["NetworkContainerType"] = createNetworkContainerRequest.NetworkContainerType
+	aiEvent.Properties["OrchestratorContext"] = fmt.Sprintf("%s", createNetworkContainerRequest.OrchestratorContext)
+
+	logger.LogNCSnapshot(aiEvent)
+}
+
+func (service *HTTPRestService) logNCSnapshots() {
+	logger.Printf("[Azure CNS] Logging periodic NC snapshots.")
```

Comment:
Is it possible to add NC count in this log statement? That would be helpeful data.

Reply:
Added.